app/com/lynxanalytics/biggraph/frontend_operations/WorkflowOperations.scala

```diff
@@ -501,6 +501,83 @@ class WorkflowOperations(env: SparkFreeEnvironment) extends ProjectOperations(en
     registerSQLOp(s"SQL$inputs", numbers.take(inputs))
   }
 
+  registerOp("Filter with SQL", "filter", category, List("input"), List("output"), new SmartOperation(_) {
+    private val input = context.inputs("input")
+    if (input.isProject) {
+      params ++= List(
+        Code("vertex_filter", "Vertex filter", language = "sql"),
+        Code("edge_filter", "Edge filter", language = "sql"))
+    } else {
```

Comment:
> Let's explicitly check for table kind here and return an error otherwise (say, for visualization). (But it's really cool that this works for tables as well! :) )

Done.